src/integrations/gitea/create_release.rs

```diff
@@ -0,0 +1,177 @@
+use crate::{
+    app_config, config,
+    dry_run::DryRun,
+    integrations::{ureq_err_to_string, CreateReleaseInput, CreateReleaseResponse},
+    state,
+    step::releases::package::{Asset, AssetNameError},
+};
+
+#[allow(clippy::too_many_arguments)]
+pub(crate) fn create_release(
+    name: &str,
+    tag_name: &str,
+    body: Option<&str>,
+    prerelease: bool,
+    gitea_state: state::Gitea,
+    gitea_config: &config::Gitea,
+    dry_run_stdout: DryRun,
+    assets: Option<&Vec<Asset>>,
+) -> Result<state::Gitea, Error> {
+    let gitea_release = CreateReleaseInput::new(tag_name, name, body, prerelease, assets.is_some());
+
+    if let Some(stdout) = dry_run_stdout {
+        gitea_release_dry_run(name, assets, &gitea_release, stdout)?;
+        return Ok(gitea_state);
+    }
+
+    let (token, agent) = initialize_state(&gitea_config.host, gitea_state)?;
+
+    let response: CreateReleaseResponse = agent
+        .post(&gitea_config.get_releases_url())
+        .query("access_token", &token)
+        .send_json(gitea_release)
+        .map_err(|source| Error::ApiRequest {
+            err: ureq_err_to_string(source),
+            activity: "creating a release".to_string(),
+            host: gitea_config.host.clone(),
+        })?
+        .into_json()
+        .map_err(|source| Error::ApiResponse {
+            source,
+            activity: "creating a release",
+            host: gitea_config.host.clone(),
+        })?;
+
+    if let Some(assets) = assets {
+        let mut upload_template = UriTemplate::new(&response.upload_url);
+        for asset in assets {
+            let file =
+                std::fs::read(&asset.path).map_err(|source| Error::CouldNotReadAssetFile {
+                    path: asset.path.clone(),
+                    source,
+                })?;
+            let asset_name = asset.name()?;
+            let upload_url = upload_template.set("name", asset_name.as_str()).build();
+            agent
+                .post(&upload_url)
+                .set("Content-Type", "application/octet-stream")
+                .set("Content-Length", &file.len().to_string())
+                .query("access_token", &token)
+                .send_bytes(&file)
+                .map_err(|source| Error::ApiRequest {
+                    err: ureq_err_to_string(source),
+                    activity: format!(
+                        "uploading asset {asset_name}. Release has been created but not published!",
+                    ),
+                    host: gitea_config.host.clone(),
+                })?;
+        }
+        agent
+            .patch(&response.url)
+            .query("access_token", &token)
+            .send_json(ureq::json!({
+                "draft": false
+            }))
+            .map_err(|source| Error::ApiRequest {
+                err: ureq_err_to_string(source),
+                activity: "publishing release".to_string(),
+                host: gitea_config.host.clone(),
+            })?;
+    }
```

Comment:
Gitea requires a `multipart/form-data` encoded request here. ureq does not support sending `multipart/form-data` requests. see: https://github.com/algesten/ureq/issues/698